### PR TITLE
tests: Add docker build test

### LINKED ISSUE #378: docker build test
add docker integration build test

### PR DESCRIPTION
**Pull request by GabyCT**
Add integration docker build test

Fixes #378

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: Dockerfiles/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2017 Intel Corporation
```
Comment (chavafg) [inline Dockerfiles/Dockerfile]: should this file be renamed to `Dockerfile.buildtest` or something similar? 
I suspect we will need to add more Dockerfiles in this directory.

File: integration/docker/build_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		runDockerCommand(0, args...)
+		Expect(ContainerExists(id)).NotTo(BeTrue())
+	})
+
+	Describe("build with docker", func() {
+		Context("docker build env vars", func() {
+			It("should display env vars", func() {
+				gopath := os.Getenv("GOPATH")
+				entirePath := filepath.Join(gopath, dockerFile)
+				args = []string{"build", "-t", name, entirePath}
```
Comment (chavafg) [inline integration/docker/build_test.go]: if the Dockerfile name is changed,  we should add `-f ` flag to specify the path of the dockerfile: E.g. `-f Dockerfile.buildtest`

File: integration/docker/build_test.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	"path/filepath"
+)
+
+const dockerFile = "src/github.com/clearcontainers/tests/Dockerfiles/."
+
+var _ = Describe("build", func() {
+	var (
+		args []string
+		id   string
+		name string = "test"
```
Comment (chavafg) [inline integration/docker/build_test.go]: maybe name this variable `image_name`

File: Dockerfiles/Dockerfile (hunk 1 above)
Comment (grahamwhaley) [inline Dockerfiles/Dockerfile]: I think +1 for that - there are dockerfiles we should add for the metrics tests for instance, and having unique names would be good (sure, that means we would then need a `docker build -f xxx` than defaulting to the dockerfile default name - but it is either that or we have a separate directory for each dockerfile.

**Comment by GabyCT** [comment]
@chavafg changes applied

**Comment by devimc** [comment]
lgtm

**Comment by chavafg** [comment]
lgtm

**Comment by grahamwhaley** [comment]
Hi,
 sorry, I'm a bit late to the party here, but seeing #405 caught my eye and made me think....
 afaik, a `docker build` does not invoke or involve the runtime at all, does it? In which case, maybe we should consider if we need this (and any other tests that do not actually involve or invoke or can be affected by) the runtime.
 I can see that maybe it gives us a little more confidence in our docker install and CI infra, but on the downside it will cost us some space, time and resource and will not actually be testing our code base.

 Thoughts @chavafg @GabyCT ?

**Pull request merged**